### ISSUE TITLE
rJava

### ISSUE THREAD

**Issue by AllisonKuhl** [hist3907b-winter2015/module4-holes#1]
For exercise 3, rJava does not seem to be working. It keeps giving me this message: 

Error : .onLoad failed in loadNamespace() for 'rJava', details:
  call: dirname(this$RuntimeLib)
  error: a character vector argument expected
Error: package or namespace load failed for ‘rJava’

Is there anyway I can fix this?

**Comment by shawngraham** [comment]
have you uncommented the line that looks like this:
#install.packages("rJava")? delete the hash, then run that line

**Comment by shawngraham** [comment]
this bit:
#install.packages('rJava')
library(rJava)

delete the hash, then run both lines. then you'll be ok.

**Comment by AllisonKuhl** [comment]
yes, and it says "Java is successfully unpacked", etc, but then it still doesn't work.

**Comment by shawngraham** [comment]
and you run the library line too, eh?

and you uncommented this: #install.packages('mallet')?

Actually, what command (what line from the script) do you run that gives you the error?

**Comment by shawngraham** [comment]
worse comes to worse, save your r script and email it to me as an attachment and I'll check it from here later tonight.

**Comment by AllisonKuhl** [comment]
I've uncommented all the necessary lines. (although, when there are two hashtags, do you delete them both or just one? I've tried both ways). 

It errors at the library(rJava) line, but when I check the library it does have rJava in it. But it doesn't seem to be recognizing it. 

Anyway, it's not a big deal, I might just do a different exercise.

**Comment by shawngraham** [comment]
Send me your script and I'll try to trouble shoot here. It's probably something fairly simple, once we see the whole thing.

Dr. Shawn Graham Carleton University @electricarchaeo

-------- Original message --------
From Allison notifications@github.com
Date: 02-28-2015 11:21 AM (GMT-05:00)
To hist3907b-winter2015/module4-holes module4-holes@noreply.github.com
Cc Shawn Graham shawn_graham@carleton.ca
Subject Re: [module4-holes] rJava (#1)

I've uncommented all the necessary lines. (although, when there are two hashtags, do you delete them both or just one? I've tried both ways).

It errors at the library(rJava) line, but when I check the library it does have rJava in it. But it doesn't seem to be recognizing it.

Anyway, it's not a big deal, I might just do a different exercise.

**Comment by AllisonKuhl** [comment]
What do you mean by script? Sorry. What I type in is just the same script as in the beginning of the exercise, this is what happens:
setwd("C:\Users\Allison.FamilyPC\Documents\Datamining\R-exercise")> # give yourself as much memory as you've got> options(java.parameters = "-Xmx5120m")> install.packages('rJava')Warning in install.packages :  downloaded length 227 != reported length 227Installing package into ‘C:/Users/Allison.FamilyPC/Documents/R/win-library/3.1’(as ‘lib’ is unspecified)trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.1/rJava_0.9-6.zip'Content type 'application/zip' length 758898 bytes (741 Kb)opened URLdownloaded 741 Kb
package ‘rJava’ successfully unpacked and MD5 sums checked
The downloaded binary packages are in   C:\Users\Allison.FamilyPC\AppData\Local\Temp\RtmpeUryJi\downloaded_packages> library(rJava)Error : .onLoad failed in loadNamespace() for 'rJava', details:  call: dirname(this$RuntimeLib)  error: a character vector argument expectedError: package or namespace load failed for ‘rJava’>   from http://cran.r-project.org/web/packages/mallet/mallet.pdfError: unexpected symbol in "  from http"> install.packages('mallet')Installing package into ‘C:/Users/Allison.FamilyPC/Documents/R/win-library/3.1’(as ‘lib’ is unspecified)trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.1/mallet_1.0.zip'Content type 'application/zip' length 4483458 bytes (4.3 Mb)opened URLdownloaded 4.3 Mb
package ‘mallet’ successfully unpacked and MD5 sums checked
The downloaded binary packages are in   C:\Users\Allison.FamilyPC\AppData\Local\Temp\RtmpeUryJi\downloaded_packages> library(mallet) Loading required package: rJavaError : .onLoad failed in loadNamespace() for 'rJava', details:  call: dirname(this$RuntimeLib)  error: a character vector argument expectedError: package ‘rJava’ could not be loaded
Date: Sat, 28 Feb 2015 13:53:30 -0800

**Comment by shawngraham** [comment]
In the screenshot below, the top panel is the 'script', ie the series of commands you want to run. The bottom window is the console where you actually run the commands and see what the output is. Does your script look like this? And did you install, in addition to R Studio, the R language as well?

![screen shot 2015-03-01 at 1 25 22 pm](https://cloud.githubusercontent.com/assets/4266026/6432155/77117a18-c016-11e4-9744-aaf3b963dab3.png)
